code/modules/power/cable.dm

```diff
@@ -511,75 +497,54 @@
-		var/disx = normal_cable.x - src.x
-		var/disy = normal_cable.y - src.y
-		if (disy == 1)
-			if (disx == 1)
-				if (normal_cable.d1 == SOUTHWEST || normal_cable.d2 == SOUTHWEST)
-					cable_surr |= NORTHEAST_UNIQUE
-			else if (disx == -1)
-				if (normal_cable.d1 == SOUTHEAST || normal_cable.d2 == SOUTHEAST)
-					cable_surr |= NORTHWEST_UNIQUE
-			else if (normal_cable.d1 == SOUTH || normal_cable.d2 == SOUTH)
-				cable_surr |= NORTH
-		else if (disy == -1)
-			if (disx == 1)
-				if (normal_cable.d1 == NORTHWEST || normal_cable.d2 == NORTHWEST)
-					cable_surr |= SOUTHEAST_UNIQUE
-			else if (disx == -1)
-				if (normal_cable.d1 == NORTHEAST || normal_cable.d2 == NORTHEAST)
-					cable_surr |= SOUTHWEST_UNIQUE
-			else if (normal_cable.d1 == NORTH || normal_cable.d2 == NORTH)
-				cable_surr |= SOUTH
-		else if (disy == 0)
-			if (disx == 1)
-				if (normal_cable.d1 == WEST || normal_cable.d2 == WEST)
-					cable_surr |= EAST
-			else if (disx == -1)
-				if (normal_cable.d1 == EAST || normal_cable.d2 == EAST)
-					cable_surr |= WEST
-		// the 'real' wires override and always connect to prevent loose ends
-		// cable_surr is any direction that needs to be connected to at all
-		// this bit does not get optimised
+	// optimises the outlier case
+		for (var/obj/cablespawner/spawner in get_step(src, EAST))
+			spawner.cable_surr |= WEST
+
+	for (var/dir_to_c in alldirs)
+	// checks for regular cables (these always connect by default)
+		var/declarer = alldirs_unique[alldirs.Find(dir_to_c)]
+		for (var/obj/cable/normal_cable in get_step(src, dir_to_c))
+			if (normal_cable.type == src.cable_type)
+				if (normal_cable.d1 == invert_dir(dir_to_c) || normal_cable.d2 == invert_dir(dir_to_c))
+					cable_surr |= declarer
 
 /// causes cablespawner to spawn cables (amazing)
 /obj/cablespawner/proc/replace()
 	var/list/directions = list()
 	if (cable_surr & NORTH)
 		directions += NORTH
-	if (cable_surr & SOUTH)
-		directions += SOUTH
-	if (cable_surr & EAST)
-		directions += EAST
 	if (cable_surr & NORTHEAST_UNIQUE)
 		directions += NORTHEAST
+	if (cable_surr & EAST)
+		directions += EAST
 	if (cable_surr & SOUTHEAST_UNIQUE)
 		directions += SOUTHEAST
+	if (cable_surr & SOUTH)
+		directions += SOUTH
+	if (cable_surr & SOUTHWEST_UNIQUE)
+		directions += SOUTHWEST
 	if (cable_surr & WEST)
 		directions += WEST
 	if (cable_surr & NORTHWEST_UNIQUE)
 		directions += NORTHWEST
-	if (cable_surr & SOUTHWEST_UNIQUE)
-		directions += SOUTHWEST
 
 	if (length(directions) == 0)
 		cable_laying(0,NORTH)
-	// multiple cables, spiral out from the centre
 	else if (src.override_centre_connection)
+	// multiple cables, spiral out from the centre 'knot'
 		for (var/i in 1 to length(directions))
 			cable_laying(0, directions[i])
+	/*
+	(this old way made knots whenever there were 3 or more way junctions)
 	else if (length(directions) >= 3)
 		for (var/i in 1 to length(directions))
-			cable_laying(0, directions[i])
+			cable_laying(0, directions[i]) */
```

Comment:
just remove it instead of commenting, git history will preserve the old version anyway